webrecorder/browsertrix
# Support additional seed URLs and custom scope type

Updates to "Seeded URL" crawl job config:
- Support setting seed URLs besides primary/start seed URL
- Only show include list ("Extra URLs") when "Custom" scope type is selected
- Moves "Page/Crawl Limits" to own tab/step and validate "Max Pages" based on number of seed URLs in URL list
- Make exclusions table collapsible
- Move `extraHops` to primary seed config
- Removed `Advanced Options` scope type divider

Resolves https://github.com/webrecorder/browsertrix-cloud/issues/493.

### Manual testing
1. Log in and go to new crawl config page
2. Choose "Seeded Crawl". Verify "Extra URLs in Scope" is not visible.
3. Choose "Custom" from "Start URL Scope". Verify "Extra URLs in Scope" is shown.
4. Enter "Extra URLs in Scope" and "Additional URLs" -> "List of URLs"
5. Go to "Limit" and enter "Max Pages". Verify that entering less than number of URLs triggers an error
6. Save config. Verify config saves and crawls as expected.
7. Go to Crawl Configs list. Verify existing crawl configs are not impacted by this change.

### Screenshots
**Seeded Crawl Scope:**
<img width="914" alt="Screen Shot 2023-02-01 at 3 28 11 PM" src="https://user-images.githubusercontent.com/4672952/216190890-ac6ca39e-b998-4d87-b12e-b6a964580aa7.png">

**Custom option in Scope Type:**
<img width="532" alt="Screen Shot 2023-02-01 at 3 31 16 PM" src="https://user-images.githubusercontent.com/4672952/216191288-c65bd2ba-449b-473c-85f9-9f3d695c553e.png">

**Crawl Limits:**
<img width="913" alt="Screen Shot 2023-02-01 at 11 47 02 AM" src="https://user-images.githubusercontent.com/4672952/216147857-d1302094-a3fb-4398-9de1-38f603e69b76.png">
